SPECS/netplan/netplan.spec

```diff
@@ -7,103 +7,241 @@
 # If this isn't defined, define it
 %{?!_systemdgeneratordir:%global _systemdgeneratordir /usr/lib/systemd/system-generators}
 
-# Force auto-byte-compilation to Python 3
-%global __python %{__python3}
+# Netplan library soversion major
+%global libsomajor 0.0
 
 
 Name:           netplan
-Version:        0.95
+Version:        0.107.1
 Release:        1%{?dist}
 Summary:        Network configuration tool using YAML
 Group:          System Environment/Base
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 License:        GPLv3
 URL:            https://netplan.io/
-# Source0:      https://github.com/canonical/%{name}/archive/%{version}/%{version}.tar.gz
-Source0:        %{name}-%{version}.tar.gz
+Source0:        https://github.com/canonical/%{name}/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
+
+# netplan build optionally depends on pyflakes, but there is a hard check for it
+# in the meson file. This patch disables that check.
+Patch0:         remove-flakes-check.patch
+
+# Some unit tests parse an openvswitch related config that requires openvswitch
+# to be installed.
+Patch1:         skip-ovs-tests.patch
+
+# Fix bug in netplan when python3-rich is not present.
+Patch2:         rich-import-failure-no-log.patch
+
+# Temporarily disabling broken test suite due to version mismatches between
+# pytest-cov and python3-coverage.
+Patch3:         disable-broken-tests.patch
 
-BuildRequires:  gcc
-BuildRequires:  make
 BuildRequires:  bash-completion-devel
-BuildRequires:  libgcc-devel
 BuildRequires:  bash-devel
-BuildRequires:  systemd-devel
+BuildRequires:  cmake
+BuildRequires:  gcc
 BuildRequires:  glib-devel
+BuildRequires:  libgcc-devel
 BuildRequires:  libyaml-devel
-BuildRequires:  util-linux-devel
+BuildRequires:  meson >= 0.61
+BuildRequires:  ninja-build
 BuildRequires:  python%{python3_pkgversion}-devel
+BuildRequires:  systemd
+BuildRequires:  systemd-devel
+BuildRequires:  systemd-rpm-macros
+BuildRequires:  util-linux-devel
 # For tests
 BuildRequires:  iproute
+BuildRequires:  libcmocka-devel
+# Blocked: not building on 3.0 yet
+# BuildRequires:  openvswitch
```

Comment:
is someone importing openvswitch?  or can we delete this as a dependency?

Reply:
For build time just the tests. On runtime it's an optional dependency to configure oven vSwitch related networking.